cpp/oneapi/dal/util/detail/load_graph.hpp

```diff
@@ -18,84 +18,176 @@
 
 #include <fstream>
 
+#include "oneapi/dal/data/detail/graph_container.hpp"
+#include "oneapi/dal/data/detail/undirected_adjacency_array_graph_impl.hpp"
 #include "oneapi/dal/data/graph_common.hpp"
 #include "oneapi/dal/data/undirected_adjacency_array_graph.hpp"
 #include "oneapi/dal/util/csv_data_source.hpp"
 #include "oneapi/dal/util/load_graph_descriptor.hpp"
+#include "services/daal_atomic_int.h"
+#include "services/daal_memory.h"
 
 namespace oneapi::dal::preview::load_graph::detail {
 
 edge_list<std::int32_t> load_edge_list(const std::string &name) {
     using int_t = std::int32_t;
-    edge_list<int_t> elist;
+
     std::ifstream file(name);
-    int_t source_vertex      = 0;
-    int_t destination_vertex = 0;
+    edge_list<int_t> elist;
+
+    char source_vertex[32], destination_vertex[32];
     while (file >> source_vertex >> destination_vertex) {
-        auto edge = std::make_pair(source_vertex, destination_vertex);
+        auto edge = std::make_pair(daal::services::daal_string_to_int(&source_vertex[0], 0),
+                                   daal::services::daal_string_to_int(&destination_vertex[0], 0));
         elist.push_back(edge);
     }
 
     file.close();
     return elist;
 }
 
-template <typename Graph>
-void convert_to_csr_impl(const edge_list<vertex_type<Graph>> &edges, Graph &g) {
-    auto layout    = oneapi::dal::preview::detail::get_impl(g);
-    using int_t    = typename Graph::vertex_size_type;
-    using vertex_t = typename Graph::vertex_type;
+typedef void (*functype)(int i, const void *a);
 
-    layout->_vertex_count = 0;
-    layout->_edge_count   = 0;
+extern "C" {
+DAAL_EXPORT void _daal_threader_for(int n, int threads_request, const void *a, functype func);
```

Comment:
Do we want to export it here? or to move to some more common place? probably it can be useful smwhere else